### PR TITLE
[On hold until Post Election] - Subscription footer

### PR DESCRIPTION
**Pull request by huwd**
Do not merge until post-election

---

A part of: A part of: https://trello.com/c/uLLaEVPV/186-dont-deploy-update-checker-results-page

---

## Includes
- Modifying the footer subscription prompt on the results page to:
  - Make it full width
  - Give it new copy
  - Change what was as govespeak insert to a simple description

## Before
- No Results ([current live](http://gov.uk/get-ready-brexit-check/results?c[]=nationality-uk))
- No Answers ([current live](http://gov.uk/get-ready-brexit-check/results))
- Lots of results([current live](https://gov.uk/get-ready-brexit-check/results?c%5B%5D=aero-space&c%5B%5D=install-service-repair&c%5B%5D=animal-ex-food&c%5B%5D=legal-service&c%5B%5D=voluntary&c%5B%5D=construction&c%5B%5D=defence&c%5B%5D=electricity&c%5B%5D=oil-gas-coal&c%5B%5D=media&c%5B%5D=sports&c%5B%5D=insurance&c%5B%5D=pharma&c%5B%5D=telecoms&c%5B%5D=consumer-goods&c%5B%5D=diamond&c%5B%5D=mining&c%5B%5D=justice-including-prisons&c%5B%5D=education&c%5B%5D=motor-trade&c%5B%5D=restaurants-catering&c%5B%5D=port-airports&c%5B%5D=postal-couriers&c%5B%5D=rail-passenger-freight&c%5B%5D=import-from-eu&c%5B%5D=export-to-eu&c%5B%5D=provide-services-do-business-in-eu&c%5B%5D=haulage-goods-across-eu-borders&c%5B%5D=ip&c%5B%5D=ip-copyright&c%5B%5D=ip-trade-marks&c%5B%5D=ip-designs&c%5B%5D=ip-patents&c%5B%5D=ip-exhaustion-rights&c%5B%5D=sell-public-sector&c%5B%5D=sell-public-sector-contracts&c%5B%5D=sell-defence-contracts&c%5B%5D=eu-uk-funding&c%5B%5D=personal-eu-org&c%5B%5D=personal-eu-org-process&c%5B%5D=personal-eu-org-use&c%5B%5D=personal-eu-org-provide&c%5B%5D=employ-eu-citizens&c%5B%5D=owns-operates-business-organisation&c%5B%5D=visiting-driving&c%5B%5D=visiting-bring-pet&c%5B%5D=visiting-ie&c%5B%5D=visiting-eu&c%5B%5D=visiting-row&c%5B%5D=travel-eu-business&c%5B%5D=working-uk&c%5B%5D=working-ie&c%5B%5D=working-eu&c%5B%5D=studying-uk&c%5B%5D=studying-ie&c%5B%5D=studying-eu&c%5B%5D=living-uk&c%5B%5D=nationality-uk))

Screenshot 2019-12-10 at 10 05 37

Screenshot 2019-12-10 at 10 05 21

## After 
- No Results ([local dev](http://finder-frontend.dev.gov.uk/get-ready-brexit-check/results?c[]=nationality-uk) | [heroku](https://finder-frontend-pr-1793.herokuapp.com/get-ready-brexit-check/results?c[]=nationality-uk))
- No Answers ([local dev](http://finder-frontend.dev.gov.uk/get-ready-brexit-check/results) | [heroku](https://finder-frontend-pr-1793.herokuapp.com/get-ready-brexit-check/results))
- Lots of results ([local dev](http://finder-frontend.dev.gov.uk/get-ready-brexit-check/results?c[]=aero-space&c[]=install-service-repair&c[]=animal-ex-food&c[]=legal-service&c[]=voluntary&c[]=construction&c[]=defence&c[]=electricity&c[]=oil-gas-coal&c[]=media&c[]=sports&c[]=insurance&c[]=pharma&c[]=telecoms&c[]=consumer-goods&c[]=diamond&c[]=mining&c[]=justice-including-prisons&c[]=education&c[]=motor-trade&c[]=restaurants-catering&c[]=port-airports&c[]=postal-couriers&c[]=rail-passenger-freight&c[]=import-from-eu&c[]=export-to-eu&c[]=provide-services-do-business-in-eu&c[]=haulage-goods-across-eu-borders&c[]=ip&c[]=ip-copyright&c[]=ip-trade-marks&c[]=ip-designs&c[]=ip-patents&c[]=ip-exhaustion-rights&c[]=sell-public-sector&c[]=sell-public-sector-contracts&c[]=sell-defence-contracts&c[]=eu-uk-funding&c[]=personal-eu-org&c[]=personal-eu-org-process&c[]=personal-eu-org-use&c[]=personal-eu-org-provide&c[]=employ-eu-citizens&c[]=owns-operates-business-organisation&c[]=visiting-driving&c[]=visiting-bring-pet&c[]=visiting-ie&c[]=visiting-eu&c[]=visiting-row&c[]=travel-eu-business&c[]=working-uk&c[]=working-ie&c[]=working-eu&c[]=studying-uk&c[]=studying-ie&c[]=studying-eu&c[]=living-uk&c[]=nationality-uk) | [heroku](https://finder-frontend-pr-1793.herokuapp.com/get-ready-brexit-check/results?c[]=aero-space&c[]=install-service-repair&c[]=animal-ex-food&c[]=legal-service&c[]=voluntary&c[]=construction&c[]=defence&c[]=electricity&c[]=oil-gas-coal&c[]=media&c[]=sports&c[]=insurance&c[]=pharma&c[]=telecoms&c[]=consumer-goods&c[]=diamond&c[]=mining&c[]=justice-including-prisons&c[]=education&c[]=motor-trade&c[]=restaurants-catering&c[]=port-airports&c[]=postal-couriers&c[]=rail-passenger-freight&c[]=import-from-eu&c[]=export-to-eu&c[]=provide-services-do-business-in-eu&c[]=haulage-goods-across-eu-borders&c[]=ip&c[]=ip-copyright&c[]=ip-trade-marks&c[]=ip-designs&c[]=ip-patents&c[]=ip-exhaustion-rights&c[]=sell-public-sector&c[]=sell-public-sector-contracts&c[]=sell-defence-contracts&c[]=eu-uk-funding&c[]=personal-eu-org&c[]=personal-eu-org-process&c[]=personal-eu-org-use&c[]=personal-eu-org-provide&c[]=employ-eu-citizens&c[]=owns-operates-business-organisation&c[]=visiting-driving&c[]=visiting-bring-pet&c[]=visiting-ie&c[]=visiting-eu&c[]=visiting-row&c[]=travel-eu-business&c[]=working-uk&c[]=working-ie&c[]=working-eu&c[]=studying-uk&c[]=studying-ie&c[]=studying-eu&c[]=living-uk&c[]=nationality-uk))

Screenshot 2019-12-10 at 10 00 35

Screenshot 2019-12-10 at 10 01 34

### CODE REVIEW

File: app/views/brexit_checker/results.html.erb [hunk 1]
```diff
@@ -76,10 +76,7 @@
 
     <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible govuk-section-break--bold">
     <div class="govuk-grid-row">
-      <div class="govuk-grid-column-one-third">
-        &nbsp; <!-- without this space, it will push the two-thirds column to the left -->
-      </div>
-      <div class="govuk-grid-column-two-thirds">
+      <div class="govuk-grid-column-full">
```
Comment (vanitabarrett) [inline app/views/brexit_checker/results.html.erb]: I don't think we need this, I think we can actually delete the surrounding `govuk-grid-column-full` and `govuk-grid-row`

File: app/views/brexit_checker/_stay_updated.html.erb [hunk 2]
```diff
@@ -3,9 +3,9 @@
 </h2>
 
 <div class="govuk-!-margin-top-4">
```
Comment (vanitabarrett) [inline app/views/brexit_checker/_stay_updated.html.erb]: This margin class isn't needed any more

File: app/views/brexit_checker/_stay_updated.html.erb [hunk 3]
```diff
@@ -1,4 +1,6 @@
-<h2 class="govuk-heading-l stay-updated__heading"><%= t("brexit_checker.stay_updated.heading") %></h2>
+<h2 class="govuk-heading-m stay-updated__heading">
```
Comment (vanitabarrett) [inline app/views/brexit_checker/_stay_updated.html.erb]: I'm not sure the `stay-updated__heading` class is being used?

File: app/assets/stylesheets/views/_brexit_checker_results_page.scss [hunk 4]
```diff
@@ -47,8 +47,12 @@
     border-bottom: none;
   }
 
-  .stay-updated__heading {
-    margin-top: govuk-spacing(6);
+  .stay-updated_description {
+    margin-bottom: govuk-spacing(4);
+    font-size: 20px;
```
Comment (vanitabarrett) [inline app/assets/stylesheets/views/_brexit_checker_results_page.scss]: You could probably replace this `font-size` and media-query with a mixin instead: @include govuk-font(19). It depends how set we are on 20px on desktop and 16px below, as I think our standard font is normally 19px on desktop.

File: app/views/brexit_checker/results.html.erb [hunk 5]
```diff
@@ -74,7 +74,7 @@
       <%= render 'results_with_no_actions', criteria: @criteria %>
     <% end %>
 
-    <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible govuk-section-break--bold">
```
Comment (vanitabarrett) [inline app/views/brexit_checker/results.html.erb]: Could we replace this with a border-top on the subscribe section instead?

File: app/views/brexit_checker/_stay_updated.html.erb (hunk 3 above)
Comment (vanitabarrett) [reply app/views/brexit_checker/_stay_updated.html.erb]: Ah, I see it's removed in a later commit. I would suggest squashing the following commits into one "Update subscribe section content" kind of commit, as I'm not sure it really needs to be broken down so much - we wouldn't really want to deploy one commit without the others:

- `Change subscription heading to medium heading`
- `Change subscription description to simple p tag`
- `Update new post-election copy for subscription section`
- `Change stay updated description css`

File: app/assets/stylesheets/views/_brexit_checker_results_page.scss (hunk 4 above)
Comment (huwd) [reply app/assets/stylesheets/views/_brexit_checker_results_page.scss]: Ah great, didn't know about `@include govuk-font(19)`.  Humm, I was running off the font-sizes in the prototype... but I would prefer to make it standard so will change that.

Does the govuk-font handle it's own responsive size change? What's out standard font size to drop it down to in the media query?

File: app/views/brexit_checker/_stay_updated.html.erb (hunk 2 above)
Comment (huwd) [reply app/views/brexit_checker/_stay_updated.html.erb]: amended!

File: app/assets/stylesheets/views/_brexit_checker_results_page.scss (hunk 4 above)
Comment (vanitabarrett) [reply app/assets/stylesheets/views/_brexit_checker_results_page.scss]: `govuk-font` is responsive - 19px drops down to 16px. You can see the standard font-sizes here: https://govuk-design-system-preview.netlify.com/styles/typography/font-size/index.html I think whichever one you specify, it drops down to the next smallest on smaller devices.

File: app/views/brexit_checker/_stay_updated.html.erb (hunk 3 above)
Comment (huwd) [reply app/views/brexit_checker/_stay_updated.html.erb]: Thanks very much, have squashed those

File: app/assets/stylesheets/views/_brexit_checker_results_page.scss (hunk 4 above)
Comment (huwd) [reply app/assets/stylesheets/views/_brexit_checker_results_page.scss]: It works! Amended that, cheers these mixins are great.

File: app/views/brexit_checker/results.html.erb (hunk 5 above)
Comment (huwd) [reply app/views/brexit_checker/results.html.erb]: done!

**Comment by huwd** [comment]
@vanitabarrett much simpler thanks for a great review.

**Pull request merged**